devdoc/CodeReviews.md

```diff
@@ -0,0 +1,55 @@
+--- | ---- | --- | -----------------
+design: | An architectural or design issue | The reviewer considers the PR has a significant problem which will affect its functionality or future extensibility | reviewer/developer redesign expected before any further changes
+function: | Incorrect/unexpected functionality implemented | The function doesn't match the description in the jira, or doesn't solve the original problem | developer expected to address issue (or discuss)
+security: | Something in the code introduces a security problem | Highly security bugs e.g. injection attacks | developer expected to discuss the issue (and then address)
+bug: | A coding issue that will cause incorrect behaviour | Likely to cause confusion, invalid results or crashes.  | developer expected to address issue
+efficiency: | The code works, but may have scaling or other efficiency issues. | Inefficiency can cause problem in some key functions and areas | developer addressing the problem (or discuss)
+discuss: | Reviewer has thought of a potential problem, but not sure if it applies | Reviewer has a concern it may be an issue, and wants to check the developer has thought about and addressed the issue | Discussion - either in the PR or offline.
+||
+style: |  Reviewer points out non-conforming code style | Makes the code hard to read | Developer to fix
+indent: | A fairly obvious indentation issue | Makes the code hard to read | Developer to fix.
+format: | Any other unusual formatting | Makes the code hard to read | Developer to fix.
+typo: | Minor typing error | Makes something (code/message/comment) harder to read | Developer to fix.
+minor: | A minor issue that could be improved. | Education (the suggestion is better for a particular reasons), or something simple to clean up at the same time as other changes | Developer recommended to fix, but unlikely to stop a merge
+picky: | A very minor issue that could be improved, but is barely worth commenting on | Education, or something to clean up at the same time as other changes | Developer discretion to fix, wouldn't stop a merge
+||
+future: | An additional feature or functionality that fits in but should be done as a separate PR. | Ensure that missing functionality is tracked, but PRs are not held up by additional requirements. | Jira created (unless trivial) and number noted in response.
+question: | Review has a question that they are not sure of the answer to | Reviewer would like clarification to help understand the code or design.  The answer may lead to further comments. | An answer to the question.
+note: | Reviewer wants to pass on some information to the contributor which they may not know | Passing on knowledge/background | contributor should consider the note, but no change expected/required
+personal: | Reviewer has an observation based on personal preference/experience | Similar to a note, but more of an opinion than fact.  | Reflect on the suggestion, but no change expected.
```

Comment:
"personal" tags sounds like it could give the review opportunity to express opinions without having to back them with good reasoning.    I feel there should always be some justification for a comment other than simply being a "personal" preference. Could a different tag such as "consider" or "improvement" encourage the reviewer to back suggestions with some reasoning?

Reply:
I have kept as personal - but clarified it a bit.  I use it for things like commenting on patterns for coding guard conditions, which I think improve the readability of the code, and the structure, but aren't part of the style guide and definitely applied through the code.